test/Microsoft.AspNet.Mvc.Testing.Test/ControllerTestHelperTest.cs

```diff
@@ -0,0 +1,198 @@
+			// Arrange
+			var controller = new TestabilityController();
+
+			var context = new ControllerTestHelperContext()
+			{
+				RequestAborted = true
+			};
+
+			// Act
+			ControllerTestHelper.Initialize(controller, context);
+			var viewResult = controller.UsingContextRequestAborted();
+
+			// Assert
+			Assert.Null(viewResult);
+
+			// Arrange
+			context = new ControllerTestHelperContext()
+			{
+				RequestAborted = false
+			};
+
+			// Act
+			ControllerTestHelper.Initialize(controller, context);
+			viewResult = controller.UsingContextRequestAborted();
+
+			// Assert
+			Assert.Equal("Request", viewResult.ViewName);
+		}
+
+		[Fact]
+		public void ControllerTestability_UsingModelStateTest()
+		{
+			// Arrange
+			var controller = new TestabilityController();
+			var model = new MyModel()
+			{
+				Property1 = "ValidModel",
+				Property2 = "InvalidModel"
+			};
+
+			// Act
+			ControllerTestHelper.Initialize(controller, null);
+			controller.ModelState.AddModelError("", "error_1");
+			var result = controller.UsngModelState(model);
+
+			// Assert
+			var viewResult = Assert.IsType<ViewResult>(result);
+			Assert.Equal(model.Property2, viewResult.ViewName);
+			Assert.NotNull(viewResult.ViewData.Model);
+			Assert.Equal(1, controller.ModelState.ErrorCount);
+		}
+
+		[Fact]
+		public void ControllerTestability_UsingFormCollectionTest()
+		{
+			// Arrange
+			var controller = new TestabilityController();
+			var formData = new Dictionary<string, string[]>() { { "formKey1", new string[] { "formKey1Value" } } };
+			var formCollection = new FormCollection(formData);
+
+			var context = new ControllerTestHelperContext()
+			{
+				RequestFormCollection = formCollection
+			};
+
+			// Act
+			ControllerTestHelper.Initialize(controller, context);
+			var result = controller.UsingContextFormCollection("formKey1");
+
+			// Assert
+			var viewResult = Assert.IsType<ViewResult>(result);
+			Assert.Equal(viewResult.ViewName, "formKey1Value");
+		}
+
+		[Fact]
+		public void ControllerTestability_UsingActionTest()
+		{
+			// Arrange
+			var controller = new TestabilityController();
+
+			var helperContext = new ControllerTestHelperContext()
+			{
+				OnUrlAction = (action, controllerName, values)
+				=>
+				{
+					Assert.Equal("redirect", action);
+					return action;
+				}
+			};
+			ControllerTestHelper.Initialize(controller, helperContext);
+
+			// Act && Assert
+			var result = controller.UsingUrlAction("redirect");
+
+			// Arrange
+			controller = new TestabilityController();
+
+			helperContext = new ControllerTestHelperContext()
+			{
+				OnUrlAction = (action, controllerName, values)
+				=>
+				{
+					Assert.Equal("controller1", controllerName);
+					Assert.Equal(null, values);
+					Assert.Equal("http", helperContext.RequestScheme);
+					Assert.Equal("host1", helperContext.RequestHost);
+					Assert.Equal("fragment1", helperContext.RequestFragment);
+					Assert.Equal("redirect", action);
+					return action;
+				}
```

Comment:
This isn't really right - why is the helperContext being validated in a test? You want to validate that route values passed to URL helper are valid, and then generate a fake URL so the action can proceed.